examples/pubsub_multithread_inproc.cpp

```diff
@@ -0,0 +1,99 @@
+#include <future>
+#include <iostream>
+#include <string>
+
+#include "zmq.hpp"
+#include "zmq_addon.hpp"
+
+void PublisherThread(zmq::context_t* ctx)
+{
+	//  Prepare publisher
+	zmq::socket_t publisher(*ctx, zmq::socket_type::pub);
+	publisher.bind("inproc://#1");
+
+	// Give the subscribers a chance to connect, so they don't lose any messages
+	std::this_thread::sleep_for(std::chrono::milliseconds(20));
+
+	while (true) {
+		//  Write three messages, each with an envelope and content
+		publisher.send(zmq::str_buffer("A"), zmq::send_flags::sndmore);
+		publisher.send(zmq::str_buffer("Message in A envelope"));
+		publisher.send(zmq::str_buffer("B"), zmq::send_flags::sndmore);
+		publisher.send(zmq::str_buffer("Message in B envelope"));
+		publisher.send(zmq::str_buffer("C"), zmq::send_flags::sndmore);
+		publisher.send(zmq::str_buffer("Message in C envelope"));
+		std::this_thread::sleep_for(std::chrono::milliseconds(100));
+	}
+}
+
+void SubscriberThread1(zmq::context_t* ctx)
+{
+	//  Prepare subscriber
+	zmq::socket_t subscriber(*ctx, zmq::socket_type::sub);
+	subscriber.connect("inproc://#1");
+
+	//  Thread2 opens "A" and "B" envelopes
+	subscriber.setsockopt(zmq::sockopt::subscribe, "A", 1);
+	subscriber.setsockopt(zmq::sockopt::subscribe, "B", 1);
```

Comment:
I think that should be `subscriber.set(zmq::sockopt::subscribe, "B");`

Reply:
Oh my bad, I misread the first time. Fixed.